core/src/main/java/net/sourceforge/jnlp/util/logging/UnixSystemLog.java

```diff
@@ -33,43 +33,54 @@
 
 package net.sourceforge.jnlp.util.logging;
 
-import net.adoptopenjdk.icedteaweb.IcedTeaWebConstants;
 import net.adoptopenjdk.icedteaweb.ProcessUtils;
 import net.adoptopenjdk.icedteaweb.logging.Logger;
 import net.adoptopenjdk.icedteaweb.logging.LoggerFactory;
+import net.adoptopenjdk.icedteaweb.resources.CachedDaemonThreadPoolProvider;
 import net.sourceforge.jnlp.util.docprovider.TextsProvider;
 
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
 
-public class UnixSystemLog implements SingleStreamLogger{
 
-    private final static Logger LOG = LoggerFactory.getLogger(UnixSystemLog.class);
+public class UnixSystemLog implements SingleStreamLogger {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UnixSystemLog.class);
+
+    private static final String PREAMBLE = "IcedTea-Web java error - for more info see itweb-settings debug options or console." +
+            " See " + TextsProvider.ITW_BUGS + " for help.\nIcedTea-Web java error manual log:\n";
+
+    private static final ExecutorService executor = CachedDaemonThreadPoolProvider.createDaemonThreadPool();
 
-    public UnixSystemLog(){
-    
-    }
-    
-    
     @Override
     public void log(String message) {
-        final String s = "IcedTea-Web java error - for more info see itweb-settings debug options or console. See " + TextsProvider.ITW_BUGS + " for help.\nIcedTea-Web java error manual log: \n" + message;
+        final String s = PREAMBLE + message;
+        final String[] ss = s.split("\\R"); // split string into lines
         try {
-            String[] ss = s.split("\\n"); //exceptions have many lines
+            CompletableFuture<Void> f = CompletableFuture.completedFuture(null);
             for (String m : ss) {
-                m = m.replaceAll("\t", "    ");
-                ProcessBuilder pb = new ProcessBuilder("logger", "-p","user.err", "--", m);
-                Process p = pb.start();
-                ProcessUtils.waitForSafely(p);
-                LOG.debug("System logger called with result of {}", p.exitValue());
+                f = f.thenRunAsync(() -> logToSyslog(m), executor);
```

Comment:
As far as I can see f is not used after the call. Therefore I assume that we could create a deamon based SingleThreadExecutor and simple call execute on that one instead of wrapping all in a future.

Reply:
I admit this code is very tricky. It could most likely bee improved.
Since this line is in a for loop `f` is used in the next iteration of the loop.
This makes sure that the next call to `logToSyslog` is only started when the previous has finished.

Making this all run in daemon threads is a good idea. I will check this.

Reply:
Oh, I just see that I already define an executor which will only use daemon threads